receiver/sqlqueryreceiver/internal/observability/observability.go

```diff
@@ -0,0 +1,88 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package observability // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/sqlqueryreceiver/internal/observability"
+
+import (
+	"context"
+	"fmt"
+
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/tag"
+)
+
+func init() {
+	err := view.Register(
+		viewAcceptedLogs,
+		viewErrorCount,
+	)
+	if err != nil {
+		fmt.Printf("Failed to register sqlquery receiver's views: %v\n", err)
+	}
+}
+
+const (
+	StartError   = "start_error"
+	CollectError = "collect_error"
+)
+
+var (
+	mAcceptedLogs = stats.Int64("receiver/accepted/log/records", "Number of log record pushed into the pipeline.", "")
+	mErrorCount   = stats.Int64("receiver/errors", "Number of errors", "")
```

Comment:
I believe we should use standard metrics for the consumers instead. These metrics seem to serve the same purpose. Please see https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/1d7655100b1bd42ab7863e70cf8abc31aac03da8/pkg/stanza/adapter/receiver.go#L123-L128 as example

Reply:
Why will refused logs never be reported?

Reply:
When doing `nextConsumer.ConsumeLogs()` ([source](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/20730/files#diff-0c9a67fda52566bc14742cf772b15a23ac8c74c66e5fc4f9b86c734b809d4a09R181)), the db call error is not available, as the error is specific to a particular query, and there can be more than one query configured for the receiver.

If you look in the logs receiver's `collect()` method ([source](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/20730/files#diff-0c9a67fda52566bc14742cf772b15a23ac8c74c66e5fc4f9b86c734b809d4a09R147)), it concurrently runs all the db queries (using a "query receiver" for each query) and passes all the collected logs to the next consumer. If an error occurs in any of the "query receivers", it is logged and dropped, and the returned log count is zero.

Not sure if this should be structured differently? I don't think each "query receiver" should run `nextConsumer.ConsumeLogs()` on their own? :thinking:

Reply:
@dmitryax I have removed the duplicate metrics, only using the `EndLogsOp` function now. Let's get the PR merged and iron out the metrics details in a next step?